### PR TITLE
RUM-813 chore: Add integration test for Session Replay

### PR DESCRIPTION
**Pull request by ncreated**
### What and why?

📦 This PR adds integration test for Session Replay. It is intended to bring coverage for critical logic of SR and SR <> RUM link. Lack of this coverage may easily lead to introducing serious regressions SR functioning. See `RUM-813` for more details.

Test characteristics:
* Single test: `SRMultipleViewsRecordingScenarioTests.testSRMultipleViewsRecordingScenario()`
* Both RUM and SR enabled.
* The test navigates through 7 view controllers by tapping the "NEXT" button in nav bar.
* Each view controller is displayed still for `~1s` and presents "rich" visual content.
* At the end, "RUM end view" is presented, to signal the end of RUM session (see: [#811](https://github.com/DataDog/dd-sdk-ios/pull/811)).

| start | → | → | → | → | → | end |
|-------|---|---|---|---|---|-----|
||||||||

SR events are too complex for neat assertions, so instead we only do broad checks:
* Assert that HTTP requests for SR follow expected multipart format.
* Assert that RUM and SR events share the same `app`, `session` and `view` IDs.
* Assert presence of main types of records in SR segment (_full_, _incremental_, _meta_, _focus_).
* Assert that number of wireframes in _full_ and _incremental_ records is within expected range.
* Assert that some _incremental_ records contain touch information.

### How?

Unlike RUM, Logs and Trace, Session Replay uses HTTP multipart requests, which required distinct implementation and addition of new utilities:
* `SRRequestMatcher` - To parse `multipart/form-data` request and extract form fields.
* `SRSegmentMatcher` - To add strict interface to generic JSON data denoting SR segments.
* `JSONObjectMatcher` and `JSONArrayMatcher` - To match values, objects and arrays in generic JSON objects (`[String: Any]`) in a convenient and type-safe way.

To don't-repeat-ourselves, the `UIViewControllers` used in SR integration test are taken from [`SRSnapshotTests`](https://github.com/DataDog/dd-sdk-ios/pull/1162) project. To enable their reuse, I added `SRFixtures` package defined under `DatadogSessionRelpay/*`. This package makes a "library of view controllers" dedicated for SR testing. It is used by both `IntegrationTests` and `SRSnapshotTests` projects.

The `HTTPServerMock` required small update to deal with SR content type headers and pass them back reliably to test runner.

To facilitate code review, here's an example of raw SR request that this PR deals with:

<p align="center">
Screenshot 2023-09-19 at 11 56 07
</p>

### Review checklist
- [x] Feature or bugfix MUST have appropriate tests (unit, integration)
- [x] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Add CHANGELOG entry for user facing changes

### Custom CI job configuration (optional)
- [ ] Run unit tests
- [ ] Run integration tests
- [ ] Run smoke tests

### CODE REVIEW

File: DatadogCore/Tests/Matchers/RUMSessionMatcher.swift [hunk 1]
```diff
@@ -74,18 +88,27 @@ internal class RUMSessionMatcher {
         fileprivate(set) var longTaskEvents: [RUMLongTaskEvent] = []
     }
 
+    /// RUM application ID for this session.
+    let applicationID: String
+    /// The ID of this session in RUM.
+    let sessionID: String
+
+    let applicationLaunchView: ViewVisit?
```
Comment (maciejburda) [inline DatadogCore/Tests/Matchers/RUMSessionMatcher.swift:96]: could have tiny description

File: DatadogSessionReplay/SRSnapshotTests/SRFixtures/Sources/SRFixtures/Fixtures.swift [hunk 2]
```diff
@@ -105,15 +71,19 @@ internal enum Fixture: CaseIterable {
         case .popups:
             return UIStoryboard.basic.instantiateViewController(withIdentifier: "Popups")
         case .swiftUI:
-            return UIHostingController(rootView: Text("Hello SwiftUI"))
+            if #available(iOS 13.0, *) {
+                return UIHostingController(rootView: Text("Hello SwiftUI"))
+            } else {
+                return ErrorViewController(message: "`.swiftUI` fixture is only available on iOS 13+")
+            }
```
Comment (maciejburda) [inline DatadogSessionReplay/SRSnapshotTests/SRFixtures/Sources/SRFixtures/Fixtures.swift:78]: nice catch!

**Review by maciejburda — APPROVED** [review]
Great stuff! Nice catch on <iOS13 build support

File: DatadogCore/Tests/Matchers/RUMSessionMatcher.swift (hunk 1 above)
Comment (ncreated) [reply DatadogCore/Tests/Matchers/RUMSessionMatcher.swift:96]: I'll keep it this way - it's a move, not addition. And IDK what was the reason for this prop.

**Pull request merged**